scipy/linalg/flapack_gen_tri.pyf.src

```diff
@@ -13,3 +13,54 @@ subroutine <prefix>gtsv(n, nrhs, dl, d, du, b, info)
     integer intent(out) :: info
 
 end subroutine <prefix>gtsv
+
+
+subroutine <prefix>gttrf(n, dl, d, du, du2, ipiv, info)
+    ! dl, d, du, du2, ipiv, info  = gttrf(dl, d, du, [overwrite_dl=0, overwrite_d=0, overwrite_du=0])
+    ! ?GTTRF computes an LU factorization of a tridiagonal matrix A
+    ! using elimination with partial pivoting and row interchanges.
+    !
+    ! The factorization has the form
+    !    A = L * U
+    !
+    ! where L is a product of permutation and unit lower bidiagonal
+    ! matrices and U is upper triangular with nonzeros in only the main
+    ! diagonal and first two superdiagonals.
+    callstatement (*f2py_func)(&n, dl, d, du, du2, ipiv, &info)
+    callprotoargument int*, <ctype>*, <ctype>*, <ctype>*, <ctype>*, int*, int*
+
+    integer intent(hide),   depend(d), check(len(d) > 2) :: n = len(d)
+    <ftype> intent(in,out,copy), depend(n), dimension(n-1) :: dl
+    <ftype> intent(in,out,copy),            dimension(n)   :: d
+    <ftype> intent(in,out,copy), depend(n), dimension(n-1) :: du
+    <ftype> intent(out),    depend(n), dimension(n-2)    :: du2
+    integer intent(out),    depend(n), dimension(n)      :: ipiv
+    integer intent(out)                                  :: info
+
+end subroutine <prefix>gttrf
+
+
+subroutine <prefix>gttrs(trans, n, nrhs, dl, d, du, du2, ipiv, b, ldb, info)
+    ! x, info  = gttrs(dl, d, du, du2, ipiv, b, trans="N", overwrite_b=0)
+    !
+    ! ?GTTRS solves one of the systems of equations:
+    ! A*X = B  or  A**T*X = B,
+    ! with a tridiagonal matrix A using the LU factorization computed
+    ! by ?GTTRF.
+    callstatement (*f2py_func)(trans, &n, &nrhs, dl, d, du, du2, ipiv, b, &ldb, &info)
+    callprotoargument char*, int*, int*, <ctype>*, <ctype>*, <ctype>*, <ctype>*, int*, <ctype>*, int*, int*
+    threadsafe
+
+    character  optional, intent(in), check(*trans=='N'||*trans=='T'||*trans=='C') :: trans = "N"
+    integer    intent(in, hide), depend(d), check(len(d) > 2) :: n = len(d)
+    integer    intent(in, hide), depend(b), check(nrhs > 0)  :: nrhs = shape(b, 1)
+    integer    intent(in, hide), depend(b), check(ldb >= MAX(1, n)):: ldb = shape(b, 0)
```

Comment:
You don't check but equate `ldb` to `MAX(1, n)` as default value.

Reply:
I don't understand the reasoning behind that. The value of `ldb` is `shape(b, 0)` we then check if it matches the requirements of the problem.  If it does not, an error is raised. 

If we were to rewrite the line to ``:: ldb = MAX(1, n)`` wouldn't that lead to errors in cases where `b` is of the wrong shape?

Reply:
Then you check the shape of b. Ldb is related to the stride of the array and causes segfaults if set to 0

Reply:
Actually I've just double-checked the documentation; you're right it should be MAX(1, n).

Reply:
Same issue for the `?tbtrs` PR and all possible ld# vars elsewhere.